include/quic/connection_ids.hpp

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include <stdexcept>
+
+extern "C"
+{
+#include <gnutls/crypto.h>
+}
+
+#include "types.hpp"
+#include "utils.hpp"
+
+namespace oxen::quic
+{
+    // Unique reference ID for each individual connection. QUIC allows for a
+    // connection to be reached at any of n number of connection ID's (see above
+    // constexpr) at a time. As a result, we key each of those QUIC CID's to the
+    // unique reference ID, to which the connection shared_ptr is keyed in the
+    // master map container
+    struct alignas(uint64_t) ReferenceID final
+    {
```

Comment:
```suggestion
    {
        using value_type = uint64_t;
```

(goes with the std::hash suggestion I'm making)